packages/react-reconciler/src/ReactFiberHooks.js

```diff
@@ -1117,6 +1128,96 @@ function updateMemo<T>(
   return nextValue;
 }
 
+function mountDeferredValue<T>(
+  value: T,
+  config: TimeoutConfig | void | null,
+): T {
+  const [prevValue, setValue] = mountState(value);
+  mountEffect(
+    () => {
+      Scheduler.unstable_next(() => {
+        const previousConfig = ReactCurrentBatchConfig.suspense;
+        ReactCurrentBatchConfig.suspense = config === undefined ? null : config;
+        try {
+          setValue(value);
+        } finally {
+          ReactCurrentBatchConfig.suspense = previousConfig;
+        }
+      });
+    },
+    [value, config],
+  );
+  return prevValue;
+}
+
+function updateDeferredValue<T>(
+  value: T,
+  config: TimeoutConfig | void | null,
+): T {
+  const [prevValue, setValue] = updateState(value);
+  updateEffect(
+    () => {
+      Scheduler.unstable_next(() => {
+        const previousConfig = ReactCurrentBatchConfig.suspense;
+        ReactCurrentBatchConfig.suspense = config === undefined ? null : config;
+        try {
+          setValue(value);
+        } finally {
+          ReactCurrentBatchConfig.suspense = previousConfig;
+        }
+      }, config);
+    },
+    [value, config],
+  );
+  return prevValue;
+}
+
+function mountTransition(
+  config: SuspenseConfig | void | null,
+): [(() => void) => void, boolean] {
+  const [isPending, setPending] = mountState(false);
+  const startTransition = mountCallback(
+    callback => {
+      setPending(true);
+      Scheduler.unstable_next(() => {
+        const previousConfig = ReactCurrentBatchConfig.suspense;
+        ReactCurrentBatchConfig.suspense = config === undefined ? null : config;
+        try {
+          setPending(false);
+          callback();
```

Comment:
Does the callback run as a potentially separate commit, or does being inside `Scheduler.next` prevent non-explicit "unbatching"?

Reply:
The intention here is to:

1. Set pending false at high priority
2. Batch setting pending to false and _doing the actual state change_ at lower priority

As a result, you'll see pending immediately turn true, but then it turn false together with the actual state change you wanted to do.